eldritch/encounters.py

```diff
@@ -129,22 +129,137 @@ def Lodge1(char):
+  nothing = events.Nothing()
+  seq = events.Sequence([
+    spend, events.PassFail(char, check, success, nothing)
+  ])
+  choice = events.BinaryChoice(char, "Cast a banishment spell?", "Yes", "No", seq, nothing)
+  return choice
+def Sanctum3(char):
+  choice = events.MultipleChoice(
+    char, "How many sanity do you want to trade for clues?", list(range(min(char.sanity, 3)+1))
+  )
+  gain = events.GainOrLoss(char, {"clues": choice}, {"sanity": choice})
+  return events.Sequence([choice, gain], char)
+def Sanctum4(char):
+  prereq = events.AttributePrerequisite(char, "dollars", 3, "at least")
+  dues = events.Loss(char, {"dollars": 3})
+  dreams = events.Loss(char, {"sanity": 2})
+  membership = events.MembershipChange(char, False)
+  decline = events.Sequence([membership, dreams], char)
+  return events.PassFail(
+    char,
+    prereq,
+    events.BinaryChoice(char, "Pay your dues?", "Spend $3", "Decline", dues, decline),
+    decline
+  )
 def Sanctum5(char):
   check = events.Check(char, "luck", -2)
   curse = events.BlessCurse(char, False)
   return events.PassFail(char, check, events.Nothing(), curse)
-
+def Sanctum6(char):
+  return events.Nothing() #TODO: A monster appears
+def Sanctum7(char):
+  prereq = events.AttributePrerequisite(char, "clues", 2, "at least")
+  check = events.Check(char, "lore", -2)
+  close = events.Nothing() # TODO: Close a gate
+  nothing = events.Nothing()
+  cost = events.Loss(char, {"clues": 2, "sanity": 1})
+  ceremony = events.PassFail(char, check, close, nothing)
+  seq = events.Sequence([cost, ceremony], char)
+  participate = events.BinaryChoice(char, "Participate in a gating ceremony?", "Yes", "No", seq, nothing)
+  return events.PassFail(char, prereq, participate, nothing)
+
+def Witch1(char):
+  ally = events.DrawSpecific(char, "allies", "Police Detective") # TODO: prereq on the ally being in the deck
+  check = events.Check(char, "lore", -1)
+  return events.PassFail(char, check, ally, events.Nothing())
 def Witch2(char):
   check = events.Check(char, "luck", -1)
   draw = events.Draw(char, "unique", 1)
   return events.PassFail(char, check, draw, events.Nothing())
+def Witch3(char):
+  check = events.Check(char, "luck", 0)
+  cond = events.Conditional(
+    char, check, "successes",
+    {
+      0: events.Loss(char, {'sanity': 3}),
+      1: events.Delayed(char),
+      2: events.Loss(char, {'stamina': 1}),
+      3: events.Gain(char, {'stamina': 3})
+    }
+  )
+  return events.Sequence([check, cond], char)
+def Witch4(char):
+  return events.Loss(char, {"sanity": 1})
+def Witch5(char):
+  #TODO: A gate and monster appear
+  return events.Nothing()
+def Witch6(char):
+  die = events.DiceRoll(char, 1)
+  gain = events.GainOrLoss(char, {"clues": die}, {"sanity": die})
+  return events.Sequence([die, gain], char)
+def Witch7(char):
+  check = events.Check(char, "will", -2)
+  spell = events.Draw(char, "spells", 1)
+  n_items_to_lose = int(len(char.possessions)//2)
```

Comment:
Ah, you can't quite do it this way. You see, the character may spend items (e.g. the lucky cigarette case, research materials) when they make the check, which means that by the time they have failed the check, they may not have the same number of items as they did when this event was created. I think an event to lose half your items is on the roadmap, where you calculate what half is inside of resolve() so as to avoid this problem. You can stick a TODO here for now.

Reply:
TODO added. I left the code to avoid breaking anything, and it's mostly not tested anyways.